### PR TITLE
Spark 3.5: Migrate tests to JUnit5 in actions directory

### PR DESCRIPTION
**Pull request by chinmay-bhat**
This PR contains the tests in spark/actions directory, now migrated to JUnit5.

Related to PR : #9341

Issue: #9086

### CODE REVIEW

File: spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestDeleteReachableFilesAction.java [hunk 1]
```diff
@@ -133,30 +132,30 @@ private void checkRemoveFilesResults(
...
-        results.deletedEqualityDeleteFilesCount());
-    Assert.assertEquals(
-        "Incorrect number of manifest lists deleted",
-        expectedManifestListsDeleted,
-        results.deletedManifestListsCount());
-    Assert.assertEquals(
-        "Incorrect number of other lists deleted",
-        expectedOtherFilesDeleted,
-        results.deletedOtherFilesCount());
+
```
Comment (nastra) [inline spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestDeleteReachableFilesAction.java:135]: unnecessary newline

File: spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestDeleteReachableFilesAction.java [hunk 2]
```diff
@@ -345,11 +345,10 @@ public void testIgnoreMetadataFilesNotFound() {
     DeleteOrphanFiles.Result result =
         sparkActions().deleteOrphanFiles(table).olderThan(System.currentTimeMillis()).execute();
 
-    Assert.assertEquals("Should delete 1 file", 1, Iterables.size(result.orphanFileLocations()));
-    Assert.assertTrue(
-        "Should remove v1 file",
-        StreamSupport.stream(result.orphanFileLocations().spliterator(), false)
-            .anyMatch(file -> file.contains("v1.metadata.json")));
+    assertThat(result.orphanFileLocations()).as("Should delete 1 file").size().isOne();
```
Comment (nastra) [inline spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestDeleteReachableFilesAction.java:348]: it's probably better to use `hasSize(..)` here and everywhere else

File: spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestExpireSnapshotsAction.java [hunk 3]
```diff
@@ -188,8 +187,10 @@ public void testFilesCleaned() throws Exception {
     ExpireSnapshots.Result results =
         SparkActions.get().expireSnapshots(table).expireOlderThan(end).execute();
 
-    Assert.assertEquals(
-        "Table does not have 1 snapshot after expiration", 1, Iterables.size(table.snapshots()));
+    assertThat(table.snapshots())
+        .as("Table does not have 1 snapshot after expiration")
+        .size()
```
Comment (nastra) [inline spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestExpireSnapshotsAction.java:192]: same as previously mentioned. please also update all the other places

File: spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestExpireSnapshotsAction.java [hunk 4]
```diff
@@ -482,10 +477,10 @@ public void testExpireOlderThanMultipleCalls() {
             .expireOlderThan(thirdSnapshot.timestampMillis())
             .execute();
 
-    Assert.assertEquals(
-        "Should have one snapshots.", 1, Lists.newArrayList(table.snapshots()).size());
-    Assert.assertNull(
-        "Second snapshot should not present.", table.snapshot(secondSnapshot.snapshotId()));
+    assertThat(table.snapshots()).as("Should have one snapshots.").size().isOne();
```
Comment (nastra) [inline spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestExpireSnapshotsAction.java:480]:
```suggestion
    assertThat(table.snapshots()).as("Should have one snapshot").hasSize(1);
```

File: spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestDeleteReachableFilesAction.java (hunk 2 above)
Comment (chinmay-bhat) [reply spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestDeleteReachableFilesAction.java:348]: Ok!

**Comment by nastra** [comment]
@chinmay-bhat can you also please include `TestCreateActions` as that's in the same directory

File: spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestRemoveOrphanFilesAction.java [hunk 5]
```diff
@@ -139,36 +138,43 @@ public void testDryRun() throws IOException, InterruptedException {
             .filter(FileStatus::isFile)
             .map(file -> file.getPath().toString())
             .collect(Collectors.toList());
-    Assert.assertEquals("Should be 3 files", 3, allFiles.size());
+    assertThat(allFiles).as("Should be 3 valid files").hasSize(3);
 
     List<String> invalidFiles = Lists.newArrayList(allFiles);
     invalidFiles.removeAll(validFiles);
-    Assert.assertEquals("Should be 1 invalid file", 1, invalidFiles.size());
+    assertThat(invalidFiles).as("Should be 1 invalid files").hasSize(1);
```
Comment (nastra) [inline spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestRemoveOrphanFilesAction.java:145]:
```suggestion
    assertThat(invalidFiles).as("Should be 1 invalid file").hasSize(1);
```

File: spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestRemoveOrphanFilesAction.java [hunk 6]
```diff
@@ -938,8 +946,8 @@ public void testRemoveOrphanFilesWithStatisticFiles() throws Exception {
         .olderThan(System.currentTimeMillis() + 1000)
         .execute();
 
-    Assertions.assertThat(statsLocation.exists()).as("stats file should exist").isTrue();
-    Assertions.assertThat(statsLocation.length())
+    assertThat(statsLocation.exists()).as("stats file should exist").isTrue();
```
Comment (nastra) [inline spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestRemoveOrphanFilesAction.java:949]:
```suggestion
    assertThat(statsLocation).as("stats file should exist").exists();
```

File: spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestRewritePositionDeleteFilesAction.java [hunk 7]
```diff
@@ -94,9 +96,7 @@ public class TestRewritePositionDeleteFilesAction extends SparkCatalogTestBase {
   private static final int SCALE = 4000;
   private static final int DELETES_SCALE = 1000;
 
-  @Parameterized.Parameters(
-      name =
-          "formatVersion = {0}, catalogName = {1}, implementation = {2}, config = {3}, fileFormat = {4}")
+  @Parameters(name = "catalogName = {0}, implementation = {1}, config = {2}, fileFormat = {3}")
```
Comment (nastra) [inline spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestRewritePositionDeleteFilesAction.java:99]: this should also include `formatVersion`, right?

File: spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestRewritePositionDeleteFilesAction.java [hunk 8]
```diff
@@ -118,45 +117,45 @@ public TestRewritePositionDeleteFilesAction(
     this.format = format;
```
Comment (nastra) [inline spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestRewritePositionDeleteFilesAction.java:117]: the constructor should be removed and the format should be part of the parameters

File: spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestRewritePositionDeleteFilesAction.java (hunk 7 above)
Comment (chinmay-bhat) [reply spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestRewritePositionDeleteFilesAction.java:99]: `formatVersion` is not provided, nor is there a variable with that name in the class. It was only added in the Parameters() name. So I removed it from the name. 

I think when this file was made from `TestPositionDeletesTable`, they simply copied the Parameters() as it is. If you see `TestPositionDeletesTable`, it also exists only in the Parameters() name and no `format` variable exists in the class.

File: spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestRewritePositionDeleteFilesAction.java (hunk 8 above)
Comment (chinmay-bhat) [reply spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestRewritePositionDeleteFilesAction.java:117]: I don't think that would be right. 
The constructor is not only setting `format`, it also calls its parent's constructor which is calling TestBaseWithCatalog's constructor, which is starting the spark session and running sql queries.

When constructors are only setting the `@Parameter` values, I have removed the constructor. See my changes in `TestRewriteManifestsAction`.

File: spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestRemoveOrphanFilesAction.java (hunk 6 above)
Comment (chinmay-bhat) [reply spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestRemoveOrphanFilesAction.java:949]: ok!

File: spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestRemoveOrphanFilesAction.java (hunk 5 above)
Comment (chinmay-bhat) [reply spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestRemoveOrphanFilesAction.java:145]: ok!

**Comment by chinmay-bhat** [comment]
adding `TestCreateActions`

File: spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestRewritePositionDeleteFilesAction.java (hunk 8 above)
Comment (nastra) [reply spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestRewritePositionDeleteFilesAction.java:117]: I just realized that it's a mistake to have any constructor in a parameterized base class. I've cleaned this up in https://github.com/apache/iceberg/pull/9368

**Comment by nastra** [comment]
I'll hold off on reviewing this PR once https://github.com/apache/iceberg/pull/9368 is in. @chinmay-bhat in the meantime you might want to rebase this PR on top of the commit from https://github.com/apache/iceberg/pull/9368 to align the tests

**Comment by chinmay-bhat** [comment]
rebased on top of #9368

File: spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestCreateActions.java [hunk 9]
```diff
@@ -117,47 +119,46 @@ public static Object[][] parameters() {
...
-  }
+  @Parameter(index = 3)
+  private String type;
 
-  @Before
-  public void before() {
+  private TableCatalog catalog;
+
+  @BeforeEach
+  public void beforeEach() {
```
Comment (chinmay-bhat) [inline spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestCreateActions.java:154]: @nastra In your implementation, the constructor in the base class goes under the before() with `@BeforeEach`.
But `@BeforeEach` overrides functions in the subclass which have the same decorator and signature.
So, when I run `TestCreateActions`, the base class before() doesnt run & the environment is not setup. 

For now, I've changed the function name to beforeEach() to highlight the behaviour, but I don't know how to improve it.

Comment (nastra) [reply spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestCreateActions.java:154]: similar to https://github.com/apache/iceberg/pull/9364 you could just override the method and call super.before()

Comment (chinmay-bhat) [reply spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestCreateActions.java]: thanks! updated

**Comment by nastra** [comment]
@chinmay-bhat can you please rebase against latest main?

**Comment by chinmay-bhat** [comment]
rebased

File: spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestCreateActions.java [hunk 10]
```diff
@@ -117,47 +119,48 @@ public static Object[][] parameters() {
...
+        "hive"
       },
       new Object[] {
         "testhadoop",
         SparkCatalog.class.getName(),
         ImmutableMap.of(
             "type", "hadoop",
-            "default-namespace", "default")
+            "default-namespace", "default"),
+        "hadoop"
```
Comment (nastra) [inline spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestCreateActions.java:139]: why is this additional parameter needed everywhere?

File: spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestCreateActions.java [hunk 11]
```diff
@@ -94,7 +95,7 @@ public class TestCreateActions extends SparkCatalogTestBase {
 
   private static final String NAMESPACE = "default";
 
-  @Parameterized.Parameters(name = "Catalog Name {0} - Options {2}")
+  @Parameters(name = "Catalog Name {0} - Options {2}")
```
Comment (nastra) [inline spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestCreateActions.java:98]: isn't this missing the `type` parameter here with index 3?

File: spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestCreateActions.java [hunk 12]
```diff
@@ -516,17 +527,16 @@ public void testProperties() throws Exception {
...
 
     for (Map.Entry<String, String> entry : expectedProps.entrySet()) {
-      Assert.assertTrue(
-          "Created table missing property " + entry.getKey(),
-          table.properties().containsKey(entry.getKey()));
-      Assert.assertEquals(
-          "Property value is not the expected value",
-          entry.getValue(),
-          table.properties().get(entry.getKey()));
+      assertThat(table.properties())
```
Comment (nastra) [inline spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestCreateActions.java:530]: these two assertions can be simplified to a single one using `containsEntry(...)`

File: spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestCreateActions.java (hunk 11 above)
Comment (chinmay-bhat) [reply spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestCreateActions.java:98]: yeah, will update this and review all other files Parameters name

**Review by nastra — APPROVED** [review]

**Pull request merged**